NugetAudit.ps1

```diff
@@ -0,0 +1,311 @@
+    $SignedStatus = $test.Status;
+    $SignatureValid = [System.Management.Automation.SignatureStatus]::Valid;
+
+    $message = "Is Signed: $SignedStatus";
+    $requirement = "Must be signed."
+
+    Test-Condition ($SignedStatus -eq $SignatureValid) $message $requirement;
+}
+
+function Get-DoesXmlDocExist ([string]$dllPath) {
+    # CONFIRM .XML DOCUMENTATION FILE EXISTS WITH EACH DLL
+    [string]$docFile = $dllPath -replace ".dll", ".xml";
+
+    $message = "XML Documentation:";
+    $requirement = "Must exist."
+    Test-Condition (Test-Path $docFile) $message $requirement;
+}
+
+function Get-IsValidPackageId([xml]$nuspecXml) {
+    $id = $nuspecXml.package.metadata.id;
+
+    $message = "Package Id: $id";
+    $requirement = "Must begin with 'Microsoft.'"
+
+    Test-Condition ($id.StartsWith("Microsoft.")) $message $requirement;
+}
+
+function Get-IsValidVersioning() {
+    #TODO
+}
+
+function Get-IsValidAuthors([xml]$nuspecXml) {
+    $authors = $nuspecXml.package.metadata.authors;
+
+    $message = "Authors: $authors";
+    $requirement = "Microsoft must be the only owner.";
+
+    Test-Condition ($authors -eq "Microsoft") $message $requirement;
+}
+
+function Get-IsValidOwners([xml]$nuspecXml) {
+    $owners = $nuspecXml.package.metadata.owners;
+    $ownersList = $owners -split ',';
+
+    $message = "Owners: $owners";
+    $requirement = "Must include Microsoft."
+    $recommendation = "Should include nuget owner."
+
+    Test-MultiCondition ($ownersList -contains "Microsoft") ($ownersList -contains $expectedOwner -and $ownersList.Length -eq 2) $message $requirement $recommendation;
+}
+
+function Get-IsValidProjectUrl([xml]$nuspecXml) {
+    $projectUrl = $nuspecXml.package.metadata.projectUrl;
+
+    $message = "Project Url: $projectUrl";
+    $requirement = "Must match expected."
+
+    Test-Condition ($projectUrl -eq $expectedProjectUrl) $message $requirement;
+}
+
+function Get-IsValidLicenseUrl([xml]$nuspecXml) {
+    $licenseUrl = $nuspecXml.package.metadata.licenseUrl;
+
+    $message = "License Url: $licenseUrl";
+    $requirement = "Must match expected."
+    $recommendation = "Should not use FWLINK."
+
+    Test-MultiCondition ($licenseUrl -eq $expectedLicenseUrl) ($licenseUrl -notlike "*fwlink*") $message $requirement $recommendation;
+}
+
+function Get-IsValidLicenseAcceptance([xml]$nuspecXml) {
+    $requireLicenseAcceptance = $nuspecXml.package.metadata.requireLicenseAcceptance;
+
+    $message = "Require License Acceptance: $requireLicenseAcceptance";
+    $requirement = "Not mandatory requirement."
+    $recommendation = "Should require.";
+
+    Test-MultiCondition ($true) ($requireLicenseAcceptance -eq $true) $message $requirement $recommendation;
+}
+
+function Get-IsValidCopyright([xml]$nuspecXml) {
+    $copyright = $nuspecXml.package.metadata.copyright;
+    $requiredCopyright = "$([char]0x00A9) Microsoft Corporation. All rights reserved.";#"© Microsoft Corporation. All rights reserved.";
```

Comment:
minor - move all the expected/required values to the top of this file to make it easier to read later.